tests/test_preprocess_cli.py

```diff
@@ -0,0 +1,207 @@
+"""Test preprocess commandline interface."""
+
+from __future__ import annotations
+
+from pathlib import Path
+import shutil
+
+from typer.testing import CliRunner
+import yaml
+
+from janus_core.cli.janus import app
+from tests.utils import assert_log_contains, clear_log_handlers, strip_ansi_codes
+
+DATA_PATH = Path(__file__).parent / "data"
+
+runner = CliRunner()
+
+
+def write_tmp_config(config_path: Path, tmp_path: Path) -> Path:
+    """
+    Fix paths in config files and write corrected config to tmp_path.
+
+    Parameters
+    ----------
+    config_path : Path
+        Path to yaml config file to be fixed.
+    tmp_path : Path
+        Temporary path from pytest in which to write corrected config.
+
+    Returns
+    -------
+    Path
+        Temporary path to corrected config file.
+    """
+    # Load config from tests/data
+    with open(config_path, encoding="utf8") as file:
+        config = yaml.safe_load(file)
+
+    # Use DATA_PATH to set paths relative to this test file
+    for file in ("train_file", "test_file", "valid_file"):
+        if file in config and (DATA_PATH / Path(config[file]).name).exists():
+            config[file] = str(DATA_PATH / Path(config[file]).name)
+
+    # Write out temporary config with corrected paths
+    tmp_config = tmp_path / "config.yml"
+    with open(tmp_config, "w", encoding="utf8") as file:
+        yaml.dump(config, file)
+
+    return tmp_config
+
+
+def test_help():
+    """Test calling `janus preprocess --help`."""
+    result = runner.invoke(app, ["preprocess", "--help"])
+    assert result.exit_code == 0
+    assert "Usage: janus preprocess [OPTIONS]" in strip_ansi_codes(result.stdout)
+
+
+def test_preprocess(tmp_path):
+    """Test MLIP preprocessing."""
+    train_path = Path("train")
```

Comment:
these are hdf5 files, maybe stuck an .h5 as extension not to confuse the users?

Reply:
This is referring to the directory they're stored in e.g.

```
ls processed_data/
statistics.json  test  train  val

ls processed_data/test/

ls processed_data/train/
train_0.h5  train_1.h5  train_2.h5  train_3.h5

ls processed_data/val/
val_0.h5  val_1.h5  val_2.h5  val_3.h5
```

Reply:
my bad misread it..